rest/base.ts

```diff
@@ -212,19 +212,22 @@ export class Base {
     session: Session,
     data: Body,
   ): T[] {
-    if (data[this.pluralName] || Array.isArray(data[this.resourceName])) {
-      return (data[this.pluralName] || data[this.resourceName]).reduce(
-        (acc: T[], entry: Body) =>
-          acc.concat(this.createInstance<T>(session, entry)),
-        [],
-      );
-    }
-
-    if (data[this.resourceName]) {
-      return [this.createInstance<T>(session, data[this.resourceName])];
-    }
+    let instances: T[] = [];
+    this.resourceNames.forEach((resourceName, index) => {
+      if (data[this.pluralNames[index]] || Array.isArray(data[resourceName])) {
```

Comment:
This scares me a little bit 😄 - should we consider changing `resourceNames` so it's something like

```ts
resourceNames = [
  {singular: 'resource', plural: 'resources'},
  {singular: 'resource2', plural: 'resources2'},
]
```

Reply:
yeah we can totally do that